presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -1044,6 +1061,19 @@ private static OGCGeometry geometryFromText(Slice input)
         return geometry;
     }
 
+    private static OGCGeometry geometryFromBinary(Slice input)
+    {
+        OGCGeometry geometry;
+        try {
+            geometry = OGCGeometry.fromBinary(input.toByteBuffer());
+        }
+        catch (IllegalArgumentException e) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid WKB: " + input.toByteBuffer(), e);
```

Comment:
Including the binary input in the error message is probably not useful, is it? Perhaps, include `e.getMessage()` instead.